cfpb/sbl-frontend
# feat(Upload - Longpolling): Clientside determination of `VALIDATION_EXPIRED`

closes #628 

## Changes (06/11/2024)
- enhancement(Upload - Longpolling): evaludation of `VALIDATION_EXPIRED` occurs on **all** get submission latest fetches -- reference time point changed back to the `submission_time` (i.e. upload time).

## Changes (06/10/2024)
- enhancement(Upload - Longpolling): evaluation of `VALIDATION_EXPIRED` occurs on **all** get submission latest fetches -- reference time point will be the **initial** get submission latest fetch

## Changes
- feat(Upload - Longpolling): Implemented clientside timeout - `VALIDATION_EXPIRED`
- env(Upload - Longpolling): added `SBL_VALIDATION_TIMEOUT_SECONDS` as an env variable

## How to test
- Set `SBL_VALIDATION_TIMEOUT_SECONDS` in your `.env` file to 4 seconds. Can be changed as necessary
- Navigate to Upload
- Upload a larger csv (e.g. 3MB or higher) such as [sblar_syntax_errors_large.csv](https://github.com/user-attachments/files/15776943/sblar_syntax_errors_large.csv)
- Verify that the alert and other visual elements align with the `VALIDATION_EXPIRED` state. (See screenshot)
- Verify that _Resolve Errors_ and _Review Warnings_ (uses Get Submission Latest) have no regressions.

## Screenshot
<img width="1426" alt="Screenshot 2024-06-07 at 3 25 14 AM" src="https://github.com/cfpb/sbl-frontend/assets/13324863/c90a6a64-b7d2-4f13-8680-c8b2c366bc98">

## Questions on the functional purpose and the user experience of this feature addition
- Potentially, we can have a scenario where the user can refresh after the time limit exceeded but the validation **succeeded**. Will this confuse the user?
- We should ask the backend to discard the result if the time limit has passed.
